### ISSUE TITLE
phantomjs 2.0.0 not supported

### ISSUE THREAD

**Issue by bbenezech** [rofrischmann/inline-style-prefixer#33]
Please note that I'm not requesting any kind of real support (it's a headless browser after all), I'm just wondering why it happened all of a sudden.

Long story short, `browserInfo.browser` is missing.

```
phantomjs -v
2.0.0
```

```
userAgent: Mozilla/5.0 (Macintosh; Intel Mac OS X) AppleWebKit/538.1 (KHTML, like Gecko) PhantomJS/2.0.0 Safari/538.1
name: PhantomJS
phantom: true
version: 2
webkit: true
x: true
browser:
osversion: NaN
prefix inline: Webkit
prefix CSS: -webkit-

Your userAgent seems to be not supported by inline-style-prefixer. Feel free to open an issue.
```

**Comment by rofrischmann** [comment]
Hey @bbenezech,
First of all inline-style-prefixer uses [bowser](https://github.com/ded/bowser) to detect browser information. bowser supports phantomjs detection which is why there is `phantom: true`, `name: PhantomJS`, etc.

But as inline-style-prefixer uses caniuse.com data to detect browser features which also includes prefix information it only supports browsers listed on caniuse.com which are listed in the README.
So while the bowser information is ready, inline-style-prefixer can't map the browser to one of the caniuse browsers which is why it actually is left out. 

If you got any proposal on how to do this better, or what to return instead of nothing, feel free to tell me :p. But as I don't know phantom by detail I can not support it.

> NOTE: Soon I will replace bowser with a self-made more recent browser detection tool that will throw if e.g. a phantom userAgent was used.

**Comment by bbenezech** [comment]
Thanks @rofrischmann.

Is there a solution to get rid of the warning? Nobody needs prefixes on PhantomJS, but the warnings are cluttering my tests a bit. :mega:

**Comment by rofrischmann** [comment]
@bbenezech It is possible to just add some valid global.userAgent? or would this crush your tests. Or perhaps you might tell me which context you're into. If it's e.g. Radium they provide some information about getting rid of the errors in their FAQ.

**Comment by bbenezech** [comment]
I guess you are refering to https://github.com/FormidableLabs/radium/tree/master/docs/faq#how-can-i-get-rid-of-useragent-warnings-in-tests

This is a different issue (no userAgent at all server-side)

Here I have a userAgent `Mozilla/5.0 (Macintosh; Intel Mac OS X) AppleWebKit/538.1 (KHTML, like Gecko) PhantomJS/2.0.0 Safari/538.1`, but since caniuse doesn't care (normal), I get a warning `Your userAgent seems to be not supported by inline-style-prefixer. Feel free to open an issue.`

Not really necessary.
Maybe you can add a whitelist of headless userAgent that should not be considered? By grepping `PhantomJS` for example?

**Comment by rofrischmann** [comment]
Oh woops, forgot about that :/ Well yeah I could at least at a whitelist or sth, but it should be as small and easy as possible. Would be nice to have some frequently used headless listed. As I know there's also often a problem using JSdom, so I might add that as well.

**Comment by rofrischmann** [comment]
Should be fixed after releasing latest commit (see #37 )

**Comment by d4goxn** [comment]
+1 for a whitelisted JSDom useragent, preferably something really simple like `"JSDom"`. For now, I'm just going to copy [that whitelisted string](https://github.com/rofrischmann/inline-style-prefixer/blob/master/test/prefixer-test.js#L18) to pretend JSDom is PhantomJS, sort of like how PJS pretends to be Firefox.
